proto/process/agent.proto

```diff
@@ -541,6 +549,53 @@ message OwnerReference {
 	string kind = 3;
 }
 
+// reference https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto#L4712
+message ServicePort {
+    string name = 1;
+    string protocol = 2;
+    int32 port = 3;
+    string targetPort = 4; // int or str
+    int32 nodePort = 5;
+    string appProtocol = 6;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto#L4933
+message ServiceSessionAffinityConfig {
+    int32 clientIPTimeoutSeconds = 1;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto#L4772
+message ServiceSpec {
+    repeated ServicePort ports = 2;
+    repeated LabelSelectorRequirement selectors = 3;
+    string clusterIP = 4;
+    string type = 5;
+    repeated string externalIPs = 6;
+    string sessionAffinity = 7;
+    string loadBalancerIP = 8;
+    repeated string loadBalancerSourceRanges = 9;
+    string externalName = 10;
+    string externalTrafficPolicy = 11;
+    int32 healthCheckNodePort = 12;
+    bool publishNotReadyAddresses = 13;
+    ServiceSessionAffinityConfig sessionAffinityConfig = 14;
+    string ipFamily = 15;
+    repeated string topologyKeys = 16;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto#L4925
+message ServiceStatus {
+    repeated string ingress = 17;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/release-1.19/staging/src/k8s.io/api/core/v1/generated.proto
+message Service {
+    Metadata metadata = 1;
+    ServiceSpec spec = 2;
```

Comment:
we probably should have done this on other messages but IMHO it'd prefer to stay consistent than having two ways of defining resources 🤔

Reply:
I'm worried this specific resource spec could move a lot faster than pod, replicasets and deployments because of cloud-provider LBs, external DNS, and other network-related settings. And as we're still early in this we could consider adopting this pattern only for resources more likely to change in the future than others?

@hkaj happy to have your thoughts on this as well

Reply:
if the unstability of the resources justifies it, it make sense to me then 👍